gravitee-apim-repository/gravitee-apim-repository-mongodb/src/main/java/io/gravitee/repository/mongodb/management/upgrade/upgrader/plan/PlanDefinitionVersionUpgrader.java

```diff
@@ -32,27 +34,31 @@ public class PlanDefinitionVersionUpgrader extends MongoUpgrader {
 
     public static final int PLAN_DEFINITION_VERSION_UPGRADER_ORDER = DashboardTypeUpgrader.DASHBOARD_TYPE_UPGRADER_ORDER + 1;
 
+    @Override
+    public String version() {
+        return "v1";
+    }
+
     @Override
     public boolean upgrade() {
         var query = new Document("definitionVersion", DefinitionVersion.V4.name());
         var projection = Projections.fields(Projections.include("_id", "definitionVersion"));
 
         var bulkActions = new ArrayList<UpdateManyModel<Document>>();
-        template
-            .getCollection("apis")
+        this.getCollection("apis")
             .find(query)
             .projection(projection)
             .forEach(v4Api -> {
                 bulkActions.add(
                     new UpdateManyModel<>(
-                        new Document("api", v4Api.getString("_id")),
-                        new Document("$set", new Document("definitionVersion", DefinitionVersion.V4.name()))
+                        Filters.and(Filters.eq("api", v4Api.getString("_id")), Filters.exists("definitionVersion", false)),
+                        Updates.set("definitionVersion", DefinitionVersion.V4.name())
```

Comment:
Don't you miss the exist filter?

Reply:
it is here
`, Filters.exists("environmentId", false)`